### PR TITLE
[RW-977] Adding documentation on load_data_table to README.

### PR DESCRIPTION
**Pull request by danqrodney**
Adding support for generating data frame vs. list vs. generator.

Adding ability to filter on concept ID columns provided by requester (which we use for filtering on Person.race, ethnicity, gender.)

Updating concept search widget to use load_data_table.

### CODE REVIEW

**Comment by danqrodney** [comment]
beforeafter

Attached screenshot shows code using materialize_cohort vs. load_data_table.

Note that we're now by default returning a Pandas data frame rather than a generator from load_data_table (though the client can request a generator or list instead.)

File: py/aou_workbench_client/data.py [hunk 1]
```diff
@@ -87,5 +110,13 @@ def load_data_table(cohort_name, table, columns=None, concept_ids=None,
     field_set = FieldSet(table_query)
     request = MaterializeCohortRequest(cohort_name=cohort_name,
                                        field_set=field_set,
+                                       status_filter=cohort_statuses,
                                        page_size=page_size)
-    return materialize_cohort(request, max_results=max_results, debug=debug)
+    generator = materialize_cohort(request, max_results=max_results, debug=debug)
+    if result_type == ResultTypes.GENERATOR:
```
Comment (calbach) [inline py/aou_workbench_client/data.py]: I didn't want to rat hole on this during the meeting, but I don't think it's worth the added complexity of a polymorphic return type to save 1/2 lines of code.

Personally I'm not convinced anyone is going to take advantage of the generator aspect unless they are familiar with the implementation details so I'd probably argue to just return a dataframe. But if you're sure this doesn't make sense as a default, perhaps add a second method like `load_data_table_df()`; should be possible to pass everything through without copy-paste via kwargs.

Comment (danqrodney) [reply py/aou_workbench_client/data.py]: OK, split this up into load_data_table (which returns a generator) and load_data_frame (which returns a DataFrame).

File: py/README.md [hunk 2]
```diff
@@ -458,7 +497,30 @@ annotation_query = AnnotationQuery(order_by=['is_obese', 'DESCENDING(review_stat
...
+from aou_workbench_client.swagger_client.models import ColumnFilter, ResultFilters
+from aou_workbench_client.cdr.model import Measurement
+
+from aou_workbench_client.data import load_data_table
+
+import pandas as pd
+
+temp_filter = ResultFilters(column_filter=ColumnFilter(Measurement.measurement_source_value, 
+                                                       value='Temper'))
+measure_df = load_data_table(cohort_name='Flu', table=Measurement,
```
Comment (calbach) [inline py/README.md]: load_data_frame?

File: py/aou_workbench_client/data.py [hunk 3]
```diff
@@ -7,8 +7,17 @@
 from aou_workbench_client.swagger_client.models.result_filters import ResultFilters
 from aou_workbench_client.cohorts import materialize_cohort
 
+import pandas as pd
+
+class ResultTypes(object):
```
Comment (calbach) [inline py/aou_workbench_client/data.py]: No longer used right?

File: py/test/cohorts_api_test.py [hunk 4]
```diff
@@ -5,7 +5,7 @@
 from aou_workbench_client.swagger_client.models import MaterializeCohortRequest
 from aou_workbench_client.swagger_client.models import TableQuery, FieldSet
 from aou_workbench_client.cdr.model import Person
-from aou_workbench_client.data import load_data_table
+from aou_workbench_client.data import load_data_table, load_data_frame, ResultTypes
```
Comment (calbach) [inline py/test/cohorts_api_test.py]: ResultTypes unused

File: py/README.md [hunk 5]
```diff
@@ -111,9 +111,45 @@ rendering.
...
+max_results | No | the maximum number of rows to return in the resulting data table; defaults to no limit (all matching rows will be returned.) Note that for large cohorts, it may take a long time to get all results; it is advisable to set `max_results` to something initially during development to ensure the results are what you are looking for.
+order_by | No | a list of column names from the table or related tables to order the results by; defaults to order by `person_id` and primary key of the specified table. (Example: `[Person.gender_concept_id, Person.person_id]`)
+page_size | No | the maximum number of result rows to fetch in a single API response when retrieving results; defaults to 1000.
+result_type | No | a string indicating the type of results to return; options are in the `aou_workbench_client.data.ResultTypes` class: `ResultTypes.GENERATOR` (a generator of results that makes API calls as needed), `ResultTypes.LIST` (all of the results loaded into memory), and `ResultTypes.DATA_FRAME` (a Pandas data frame containing all of the results in memory.) Defaults to `ResultTypes.DATA_FRAME`.
```
Comment (calbach) [inline py/README.md]: Regenerate?

Comment (danqrodney) [reply py/README.md]: Removed. (This isn't generated, FYI.)

File: py/README.md (hunk 2 above)
Comment (danqrodney) [reply py/README.md]: Done.

File: py/aou_workbench_client/data.py (hunk 3 above)
Comment (danqrodney) [reply py/aou_workbench_client/data.py]: Removed.

File: py/test/cohorts_api_test.py (hunk 4 above)
Comment (danqrodney) [reply py/test/cohorts_api_test.py]: Removed.

**Pull request merged**